File: app/src/main/java/com/example/rickmortymvvm/util/observer/MutableObservable.java
package com.example.rickmortymvvm.util.observer;

import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListState;
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListAction;

public interface MutableObservable extends Observable {
    void update(PresentationCharacterListState state);

    void update(PresentationCharacterListAction action);
}

File: app/src/main/java/com/example/rickmortymvvm/util/observer/MutableObservable.kt
package com.example.rickmortymvvm.util.observer

import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListAction
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListState

interface MutableObservable : Observable {
    fun update(state: PresentationCharacterListState)

    fun update(action: PresentationCharacterListAction?)
}